source/AI.cpp

```diff
@@ -2766,9 +2766,44 @@ void AI::DoSurveillance(Ship &ship, Command &command, shared_ptr<Ship> &target)
 		unsigned total = targetShips.size() + targetPlanets.size() + targetSystems.size();
 		if(!total)
 		{
-			// If there is nothing for this ship to scan, have it hold still
-			// instead of drifting away from the system center.
-			Stop(ship, command);
+			// If there is nothing for this ship to scan, have it patrol the entire system
+			// instead of drifting or stopping.
+			// Also allows the ship to land.
+			double radius = ship.GetSystem()->ExtraHyperArrivalDistance();
+
+			// The ship is outside of the effective range of the system,
+			// so we turn it around.
+			if(ship.Position().LengthSquared() > radius * radius)
+			{
+				// Allow ships to land after a while, otherwise they would continue to accumulate in the system.
```

Comment:
so if there is nowhere to land and nothing to scan they'll just accumulate? shouldnt they change system then?
also I never really noticed the ships stopping really but mb I'm missing smth, good change anyhow

Reply:
They already change systems, if they can.

Reply:
okay so this only is something for ships that cant hyperspace, would be good to specify that in the title or smth

Reply:
It's in a comment you gave a thumbs up to four months ago.

Reply:
~~before India was another me~~